### DISCUSSION TITLE
In addition to the compiler and cmake version, what else can cause the ABI numbers of vcpkg-cmake and vcpkg-cmake-config to be inconsistent?

### DISCUSSION THREAD

**Discussion by QiuYilin** [microsoft/vcpkg#34720] [Q&A]
I have two vcpkgs with the same commit id on two different computers. The cache file generated by PC A is copied to the cache folder on PC B, but PC B cannot use the cache. The reason is vcpkg-cmake and vcpkg-cmake-config. The ABI is different.
![different_ABI](https://github.com/microsoft/vcpkg/assets/26407388/7c99ef83-9ff6-4953-af8f-3a37f3a5175c)

I checked vcpkg env,vcpkg version and compiler version, they are also same.
![image](https://github.com/microsoft/vcpkg/assets/26407388/e521cde7-c722-40c4-8471-e956935fef81)
![image](https://github.com/microsoft/vcpkg/assets/26407388/729818c6-dc25-4281-9de3-55c583683b07)
![image](https://github.com/microsoft/vcpkg/assets/26407388/df39f3ff-af39-4a20-a036-637969f7198a)
Is there anything else that I'm not aware of that affects ABI?

**Comment by Osyotr** [comment]
https://github.com/microsoft/vcpkg/issues/26346#issuecomment-1319244766

Specifically,
> 3. Use the same version of vcpkg-scripts on every machine (e.g. vcpkg repo is checked out at the same commit)

**Reply by QiuYilin** [reply]
Same vcpkg repo commit？ What if I could make sure the script didn't change? Is there a way to avoid this? This is too strict.

**Reply by Osyotr** [reply]
Sorry, my advice applies only to manifest mode. In classic mode packages are not rebuilt if new version is available and it might lead to cache misses. You might want to try `vcpkg upgrade` before sharing the cache.
